diff --git a/folly/memory/UninitializedMemoryHacks.h b/folly/memory/UninitializedMemoryHacks.h
index 9babda4..6e2f505 100644
--- a/folly/memory/UninitializedMemoryHacks.h
+++ b/folly/memory/UninitializedMemoryHacks.h
@@ -105,6 +105,9 @@ template <typename T>
   requires std::is_trivially_destructible_v<T>
 inline void resizeWithoutInitializationImpl(
     std::basic_string<T>& s, std::size_t n) {
+#if defined(_MSVC_STL_UPDATE) && _MSVC_STL_UPDATE >= 202206L
+    s.resize(n);
+#else
   if (n <= s.size()) {
     s.resize(n);
   } else {
@@ -115,6 +118,7 @@ inline void resizeWithoutInitializationImpl(
     }
     unsafeStringSetLargerSize(s, n);
   }
+#endif // defined(_MSVC_STL_UPDATE) && _MSVC_STL_UPDATE >= 202206L
 }
 
 /**
@@ -316,8 +320,11 @@ struct MakeUnsafeStringSetLargerSize {
 } // namespace folly
 
 #if defined(FOLLY_DECLARE_STRING_RESIZE_WITHOUT_INIT)
+#if defined(_MSVC_STL_UPDATE) && _MSVC_STL_UPDATE >= 202206L
+#else
 FOLLY_DECLARE_STRING_RESIZE_WITHOUT_INIT(char)
 FOLLY_DECLARE_STRING_RESIZE_WITHOUT_INIT(wchar_t)
+#endif // defined(_MSVC_STL_UPDATE) && _MSVC_STL_UPDATE >= 202206L
 #endif
 
 namespace folly {
