diff --git a/CMakeLists.txt b/CMakeLists.txt
index c48d560..2f88099 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -135,14 +135,17 @@ endif ()
 
 # Find required dependencies for thrift/lib
 if (THRIFT_LIB_ONLY OR build_all)
-  find_package(Gflags REQUIRED)
-  find_package(Glog REQUIRED)
+  find_package(gflags CONFIG REQUIRED)
+  set(LIBGFLAGS_LIBRARY gflags::gflags)
+  find_package(glog CONFIG REQUIRED)
+  set(GLOG_LIBRARIES glog::glog)
   if (THRIFT_RPC)
     find_package(fizz CONFIG REQUIRED)
     find_package(wangle CONFIG REQUIRED)
   endif ()
   find_package(ZLIB REQUIRED)
-  find_package(Zstd REQUIRED)
+  find_package(zstd CONFIG REQUIRED)
+  set(ZSTD_LIBRARIES zstd::libzstd)
   find_package(Xxhash REQUIRED)
   if (THRIFT_RPC)
     find_package(mvfst CONFIG REQUIRED)
diff --git a/thrift/cmake/FBThriftConfig.cmake.in b/thrift/cmake/FBThriftConfig.cmake.in
index 9aee56b..4d3616b 100644
--- a/thrift/cmake/FBThriftConfig.cmake.in
+++ b/thrift/cmake/FBThriftConfig.cmake.in
@@ -45,8 +45,15 @@ if (NOT Xxhash_FOUND)
   return()
 endif()
 
-find_dependency(ZLIB REQUIRED)
-find_package(mvfst CONFIG)
+find_dependency(ZLIB)
+find_dependency(mvfst CONFIG)
+find_dependency(fizz CONFIG)
+find_dependency(fmt CONFIG)
+find_dependency(folly CONFIG)
+find_dependency(gflags CONFIG)
+find_dependency(glog CONFIG)
+find_dependency(wangle CONFIG)
+find_dependency(zstd CONFIG)
 
 if (NOT TARGET FBThrift::thriftcpp2)
   include("${FBTHRIFT_CMAKE_DIR}/FBThriftTargets.cmake")
